### PR TITLE
#946 Add `--url` option for `khal new` and `{url}` for format in command line

### PR DESCRIPTION
**Pull request by ephase**
This is a fist attempt to add `URL=` in a VEVENT with khal. you can add it only in command line with `--url` option with `khal new`.

You can display URL with `{url}` in format (both in khal and ikhal).

## Todo
- [x] Add option for khal
- [x] Add an `{url}` element in format
- [x] Add possibility to edit (or add) url with `khal edit`
- [x] ~~Verify URL before save event, throw an error if not correct~~
- [x] Integrate an URL widget in ikhal

### CODE REVIEW

**Comment by ephase** [comment]
I wonder if URL verification is a good idea because URL verification is tricky, with a big false positive risk.

**Comment by ephase** [comment]
Rebase today and decide to not implement URL verification.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -13,6 +13,7 @@ may want to subscribe to `GitHub's tag feed
 not released
 
 * FIX `khal interactive` now accepts -a/-d options (as documented)
+<<<<<<< HEAD
```
Comment (WhyNotHugo) [inline CHANGELOG.rst:16]: Mismerge:

```suggestion
```

Comment (ephase) [reply CHANGELOG.rst]: Sorry for the mistake, line removed

File: CHANGELOG.rst [hunk 2]
```diff
@@ -21,6 +21,7 @@ not released
 * NEW Possibility to add a blank line before day in `khal` with
    `blank_line_before_day` option
 * FIX `new` keybinding in search no longer crash `ikhal`
+* NEW add url input for `khal` and `ikhal`
```
Comment (d7415) [inline CHANGELOG.rst:24]:
```suggestion
* NEW Add url input for `khal` and `ikhal`
```

**Review by d7415 — CHANGES_REQUESTED** [review]
Sorry this has taken so long - I started then got busy and forgot about it. I've suggested a few minor changes to the text. The code looks ok and worked on my basic testing.

It's worth thinking whether there's a need for a test here.

**Comment by ephase** [comment]
@d7415 No problem, I'm really busy too here and don't have time to participate as I want to... For test, you talk about write some pytest? Or test for user input in khal / ikhal?

**Comment by d7415** [comment]
I've added some tests (and a tweak so that setting the URL to `""` removes the field altogether) and pushed to feature/946 in this repo - GitHub doesn't want to let me push to yours. If you're happy with them and can merge them into your branch to update the PR I think we're good to go.

**Comment by d7415** [comment]
I'm guessing @ephase is still busy. If I don't hear anything by the end of the week I'll just merge feature/946 from pimutils/khal.

**Comment by ephase** [comment]
hi @d7415, sorry for not giving news here, but I'm really busy at the moment. I reviewed change and all seems OK. So I merged it to my branch.

**Comment by d7415** [comment]
Thanks!

**Pull request merged**